tests/Integration/Behaviour/Features/Scenario/Order/duplicate_order_cart.feature

```diff
@@ -0,0 +1,18 @@
+@reset-database-before-feature
+Feature: Duplicate order cart
+  In order to create order with existing order cart duplicate
+  As a Back Office (BO) user
+  I need to be able to duplicate chosen order cart
+
+  Background:
+    Given email sending is disabled
+    And the current currency is "USD"
+    And there is existing order with id 1
```

Comment:
Maybe we dont need to check the very existence as the next step will check it anyway ?

We try to make behat tests as close as possible to real world usecases. It's not easy because our tests are most of the time related to CRUD operations, but the idea is to say "A user could describe this scenario as a feature to implement into a Ecommerce CMS".

So
`As a User, when I delete order "ABC123" then order "ABC123" has status "deleted"`
quite works while
`As a User, when I delete order with ID 123 then order 123 has status 5`
is more technical-oriented

It's very hard to do in our case, but we can try to walk this way by avoiding some mathematical checks such as "object X exists" and rather checks "object X name is 'red mug'" 😉

Reply:
@matks I totally agree to this and would like to do it more like user behavior when it comes to Behat, just the way I understand it getting orders by reference "ABC123" would be better but then it goes out of the Domain scope - uses legacy Object Model class as the current GetOrderForViewing is not capable to get order by reference and expects id (int type).

It would be really nice to find user oriented way to test code components with Behat.

For the current state of the Domain code written I see several ways to do that:
1. Test code components with technical oriented style but also write user behavior tests which interact with the UI - as Behat is designed for user behavior. OR
2. Improve our Domain code components to be able to use references like "ABC123.

Would be nice if we could find nice solution to this :)

And yes, we don't need to check the very existence